### PR TITLE
feat: support config_name in all JumpStart interfaces

### PR DESCRIPTION
**Pull request by Captainia**
*Issue #, if available:*

*Description of changes:*

The second part of the implementation that passes in `config_name` to all of the JumpStart interfaces, and also support set config in the JumpStartModel class.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/accept_types.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def retrieve_default(
     tolerate_deprecated_model: bool = False,
     sagemaker_session: Session = DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
     model_type: JumpStartModelType = JumpStartModelType.OPEN_WEIGHTS,
+    config_name: Optional[str] = None,
```
Comment (evakravi) [inline src/sagemaker/accept_types.py:80]: in the future, we should think of a way to consolidate all these JS related fields perhaps into a dataclass or kwargs, so we don't need to update all these function prototypes whenever a new feature is added.

**Review by makungaj1 — COMMENTED** [review]
Could we get this fix??
https://github.com/aws/sagemaker-python-sdk/blob/fbc484cd2a057711127aeecac46f37f7df5f3c61/src/sagemaker/jumpstart/utils.py#L959

This should return `Dict[str, JumpStartMetadataConfig]`

**Review by makungaj1 — COMMENTED** [review]
This one also.

https://github.com/aws/sagemaker-python-sdk/blob/fbc484cd2a057711127aeecac46f37f7df5f3c61/src/sagemaker/jumpstart/types.py#L1083

Should be `benchmark_metrics: Dict[str, List[JumpStartBenchmarkStat]]`

**Comment by Captainia** [comment]
> Could we get this fix??
> 
> https://github.com/aws/sagemaker-python-sdk/blob/fbc484cd2a057711127aeecac46f37f7df5f3c61/src/sagemaker/jumpstart/utils.py#L959
> 
> This should return `Dict[str, JumpStartMetadataConfig]`

These (and the other one) are already fixed in the current PR, can you double check?

File: src/sagemaker/accept_types.py [hunk 2]
```diff
@@ -98,6 +99,7 @@ def retrieve_default(
             object, used for SageMaker interactions. If not
             specified, one is created using the default AWS configuration
             chain. (Default: sagemaker.jumpstart.constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION).
+        config_name (Optional[str]): Name of the JumpStart Model config to apply. (Default: None).
```
Comment (evakravi) [inline src/sagemaker/accept_types.py:102]: check with @judyheflin  about how to describe this

**Review by evakravi — APPROVED** [review]

**Review by liujiaorr — APPROVED** [review]

**Pull request merged**